### ISSUE TITLE
`IndexedSemantics` unexpectedly merges the semantics of its descendants

### ISSUE THREAD

**Issue by valin4tor** [flutter/flutter#84947]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update `lib/main.dart` as follows:

```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Welcome to Flutter',
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Welcome to Flutter'),
        ),
        body: ListView(
          children: [
            Row(children: [
              Text('Item 1'),
              Text('Item 2'),
            ]),
            Row(children: [
              Text('Item 3'),
              Text('Item 4'),
            ]),
          ],
        ),
      ),
    );
  }
}
```

3. Run the app in iOS Simulator or on a real iPhone

If running on the iOS Simulator:
4. Use the macOS Accessibility Inspector and inspect one of the items

If running on a real iPhone:
4. Enable VoiceOver and focus one of the items

**Note:** I haven't yet tested on an Android device or on other platforms.

**Expected results:** 

Each item is semantically unique, as below (when using `Column` instead of `ListView`):

![image](https://user-images.githubusercontent.com/4248177/122688431-0af5ec80-d214-11eb-8b58-950f70c0d5f1.png)

**Actual results:** 

Each child of the `ListView` is semantically merged, as if wrapped with `MergeSemantics`:

![image](https://user-images.githubusercontent.com/4248177/122688467-41336c00-d214-11eb-9918-7a34411fa0a1.png)

This only occurs when `addSemanticIndexes` is set to `true`, the default value. This causes each child to be wrapped with a `IndexedSemantics` widget, which seems to behave like `MergeSemantics`, in addition to its intended behaviour.

<details>
  <summary>Logs</summary>

https://gist.github.com/davidbailey00/7a3aa2775b9bad81db30837a2726dea1

```
% flutter analyze                                                                                                               ✹
Analyzing bug...
No issues found! (ran in 1.7s)
```

```
% flutter doctor -v                                                                                                             ✹
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-arm, locale en-GB)
    • Flutter version 2.2.2 at /Users/david/Applications/flutter
    • Framework revision d79295af24 (9 days ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/david/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment Zulu11.48+21-CA (build 11.0.11+9-LTS)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • iPhone SE (2nd generation) (mobile) • BA3C196A-2773-45B3-91F1-2BBD223AF1F4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)                        • chrome                               • web-javascript • Google Chrome 91.0.4472.114

! Doctor found issues in 1 category.
```

</details>

**Comment by davidbailey00** [comment]
Upon further inspection, this issue only occurs when `addSemanticIndexes` is set to `true` (the default value).

**Comment by davidbailey00** [comment]
I believe this bug lies in the `IndexedSemantics` widget and have updated the issue title accordingly.

**Comment by darshankawar** [comment]
On iPhone 6s:

<img width="669" alt="Screenshot 2021-06-21 at 5 46 36 PM" src="https://user-images.githubusercontent.com/67046386/122760354-a6409d80-d2b8-11eb-82de-9fa8cdef3ef9.png">

<details>
<summary> flutter doctor -v </summary>

```
dhs@Dhss-MacBook-Pro ncflutter % flutter doctor -v
[✓] Flutter (Channel master, 2.3.0-17.0.pre.414, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.3.0-17.0.pre.414 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 757c3add14 (7 hours ago), 2021-06-20 22:29:02 -0400
    • Engine revision 9520bb15b3
    • Dart version 2.14.0 (build 2.14.0-228.0.dev)

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 91.0.4472.106

• No issues found!

```

</details>

**Comment by dkwingsmt** [comment]
cc @chunhtai

**Comment by chunhtai** [comment]
not I don't think it is IndexedSemantics is at fault here. Some where in the scaffold probably has explicitChildNode: true.

You can achieve your goal by doing this
```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Welcome to Flutter',
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Welcome to Flutter'),
        ),
        body: ListView(
          children: [
            Semantics(explicitChildNode:true, child: Row(children: [
              Text('Item 1'),
              Text('Item 2'),
            ])),
            Semantics(explicitChildNode:true, child: Row(children: [
              Text('Item 3'),
              Text('Item 4'),
            ])),
          ],
        ),
      ),
    );
  }
}
```

**Comment by chunhtai** [comment]
closing as WAI
